File: output/StarCoder/codenet/C++/Python/s208096306.py
import sys

def gcd(a, b):
    if a < b:
        a, b = b, a
    r = a % b
    while r > 0:
        a = b
        b = r
        r = a % b
    return b

n = int(input().strip())

ans = int(input().strip())

for i in range(1, n):
    a = int(input().strip())
    ans = gcd(ans, a)

print(ans)

File: output/StarCoder/codenet/C++/Java/s208096306.java
import java.util.*;

public class s208096306 {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int ans = sc.nextInt();
        for (int i = 1; i < n; i++) {
            int a = sc.nextInt();
            ans = gcd(ans, a);
        }
        System.out.println(ans);
    }

    public static int gcd(int a, int b) {
        if (a < b) {
            int temp = a;
            a = b;
            b = temp;
        }
        int r;
        while ((r = a % b) > 0) {
            a = b;
            b = r;
        }
        return b;
    }
}